cmd/kyma/alpha/uninstall/cmd.go

```diff
@@ -92,15 +135,67 @@ func (cmd *command) Run() error {
 		}
 	}
 
-	installer, err := deployment.NewDeployment(installationCfg, deployment.Overrides{}, cmd.K8s.Static(), updateCh)
+	installer, err := deployment.NewDeployment(installCfg, deployment.Overrides{}, cmd.K8s.Static(), updateCh)
 	if err != nil {
 		return err
 	}
 
-	err = installer.StartKymaUninstallation()
-	if err != nil {
-		return err
+	uninstallErr := installer.StartKymaUninstallation()
+
+	if err := cmd.deleteComponentsListFile(installCfg.ComponentsListFile); err != nil {
+		return errors.Wrap(err, uninstallErr.Error())
+	}
+
+	if uninstallErr == nil {
+		cmd.showSuccessMessage()
 	}
+	return uninstallErr
+}
 
+func (cmd *command) recoverComponentsListFile(file string, data []byte) error {
+	restoreClStep := stepFactory.AddStep("Restore component list used for initial Kyma installation")
+	err := ioutil.WriteFile(file, data, 0600)
+	if err == nil {
+		restoreClStep.Success()
+	} else {
+		restoreClStep.Failure()
+	}
+	return err
+}
+
+func (cmd *command) deleteComponentsListFile(file string) error {
+	if _, err := os.Stat(file); os.IsNotExist(err) {
+		// file doesn't exist
+		return nil
+	}
+	if err := os.Remove(file); err != nil {
+		return err
+	}
 	return nil
 }
+
+func (cmd *command) retrieveKymaMetadata() (*metadata.KymaMetadata, error) {
+	getMetaStep := stepFactory.AddStep("Retrieve Kyma metadata")
+	provider := metadata.New(cmd.K8s.Static())
+	metadata, err := provider.ReadKymaMetadata()
+	if err == nil {
+		if metadata.Version == "" {
+			getMetaStep.Failure()
+			return metadata, fmt.Errorf("No Kyma installation found")
+		}
+		getMetaStep.Successf("Kyma was installed from source '%s'", metadata.Version)
+	} else {
+		getMetaStep.Failure()
+	}
+	return metadata, err
+}
```

Comment:
If we get more metadata-related logic, I'd consider factoring it out in smth like a `MetadataClient` to share with e.g. the version command.

Reply:
Isn't the provider already a kind of client?